addons/point_of_sale/models/pos_category.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models, tools, _
+
+
+class PosCategory(models.Model):
+    _name = "pos.category"
+    _description = "Public Category"
+    _order = "sequence, name"
+
+    _constraints = [(models.Model._check_recursion, _('Error ! You cannot create recursive categories.'), ['parent_id'])]
```

Comment:
Hey man i think you forget some thing,
```
@api.constrains('parent_id')
def _check_recursion(self):
    if not super(PosCategory, self)._check_recursion():
        raise(_('Error ! You cannot create recursive categories.'))
```

![forget](https://cloud.githubusercontent.com/assets/14940179/12125525/e2332540-b3a8-11e5-88f2-48e6ad5577ae.gif)
I like your clone :laughing: